### DISCUSSION TITLE
OIDC auth with v1.9

### DISCUSSION THREAD

**Discussion by drustan** [peering-manager/peering-manager#893] [General]
Hello guys,  

I finally got around to upgrading to version 1.9, but I’m running into issues with OIDC authentication—it’s not working anymore.  

### What Worked Before  

In version 1.8, I was using `mozilla-django-oidc` with the following setup:  

**Environment Variable:**  
`REMOTE_AUTH_ENABLED=true`  

**`oidc_config.py`:**  
```python  
OIDC_RP_CLIENT_ID = "peering-manager"  
OIDC_RP_CLIENT_SECRET = "XXXXXXXXXXXXXXXXXXX"  

OIDC_RP_SIGN_ALGO = "RS256"  
OIDC_OP_JWKS_ENDPOINT = "https://keycloak.example.net/realms/example/protocol/openid-connect/certs"  

OIDC_OP_AUTHORIZATION_ENDPOINT = "https://keycloak.example.net/realms/example/protocol/openid-connect/auth"  
OIDC_OP_TOKEN_ENDPOINT = "https://keycloak.example.net/realms/example/protocol/openid-connect/token"  
OIDC_OP_USER_ENDPOINT = "https://keycloak.example.net/realms/example/protocol/openid-connect/userinfo"  

LOGIN_REDIRECT_URL = "https://peering-manager.example.net/"  
LOGOUT_REDIRECT_URL = "https://peering-manager.example.net/"  

OIDC_CREATE_USER = True  
```  

This setup worked fine, and the login page showed the SSO button.  

### The Problem  

After upgrading to v1.9, the SSO button doesn’t appear anymore. I followed the new docs, removed the `oidc_config.py` file, and switched to using `social-auth-core`/`social-auth-app-django`. Here’s what I’ve got now:  

**Environment Variables:**  
```bash  
REMOTE_AUTH_ENABLED=true  
REMOTE_AUTH_BACKEND=social_core.backends.open_id_connect.OpenIdConnectAuth  
SOCIAL_AUTH_BACKEND_ATTRS={"oidc": ("SSO", "fa-fw fa-brands fa-openid")}  
SOCIAL_AUTH_OIDC_ENDPOINT=https://keycloak.example.net/realms/example  
SOCIAL_AUTH_OIDC_OIDC_ENDPOINT=https://keycloak.example.net/realms/example  
SOCIAL_AUTH_OIDC_KEY=peering-manager  
SOCIAL_AUTH_OIDC_SECRET=XXXXXXXXXXXXXXXXXXXXXX  
SOCIAL_AUTH_OIDC_SCOPE=["openid", "profile", "email", "roles"]  
LOGOUT_REDIRECT_URL=https://peering-manager.example.net/  
```  

Still no luck.  

### What Else I Tried  

I also gave the Keycloak backend a shot with this setup:  

**Keycloak Config:**  
```bash  
REMOTE_AUTH_ENABLED=true  
REMOTE_AUTH_BACKEND=social_core.backends.keycloak.KeycloakOAuth2  
SOCIAL_AUTH_KEYCLOAK_KEY="peering-manager"  
SOCIAL_AUTH_KEYCLOAK_SECRET=XXXXXXXXXXXXXXXXXXXXXX  
SOCIAL_AUTH_KEYCLOAK_PUBLIC_KEY=MIIBIjANBgkqhkiG9w0BSQEFAAOCAQ8AMIIBCgKCAQEAu61rLzUBVFll7y0ErzZhqOCnLmSoYmElHGriOtGDJRD18ND3ajER5aWVr0WW2z4b/wZCVQ6g/CuEkfkHVTnoRCPpwImAwqZgvZjgAHv+Ucq7drtuuuyHKoxaNLIX8n+ViIBbo9w9c19H44gdRINonb32aWQukcbK0pr6i2CUsVCCcjWLUH40MzNV+DVuadRlghGai9oHD7xUNqNHabBuAIvQ4efx7nMk/YVmoSmpeKa8gKXwA1UObaJTmSSwVw5fEGgoTRsIyQcVXGpCLJgFGrBw0Yz8j2fpv/omC7+QfHr01aXwjx9CN9EKlsdNw7ZJ6xTprRSqbqgmuUO0oOoQRQIDAQAB  
SOCIAL_AUTH_KEYCLOAK_ACCESS_TOKEN_URL="https://keycloak.example.net/realms/example/protocol/openid-connect/token"  
SOCIAL_AUTH_JSONFIELD_ENABLED=true  
SOCIAL_AUTH_KEYCLOAK_AUTHORIZATION_URL="https://keycloak.example.net/realms/example/protocol/openid-connect/auth"  
```  

But still nothing works.  

### Any Ideas?  

Am I missing something obvious? Would really appreciate any pointers here!  

Thanks for your help!

**Comment by gmazoyer** [comment]
Since you mention environment variables I suppose you use a piece of code to load them or maybe you are using the Docker image built from [this repository](https://github.com/peering-manager/docker). If that's the case, using env vars won't work as it's close to impossible to bind every variables in the [file in charge of the configuration](https://github.com/peering-manager/docker/blob/main/configuration/configuration.py).

However you should be able to mount a Python file in the container, under `/etc/peering-manager/config/` with your configuration for SSO in it.

**Reply by drustan** [reply]
Hi @gmazoyer,

I’m indeed using the Docker image, and that was the issue. Thanks for pointing it out!
